x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/step_rule_actions/index.tsx

```diff
@@ -34,6 +42,15 @@ const stepActionsDefaultValue = {
 
 const GhostFormField = () => <></>;
 
+const getThrottleOptions = (throttle?: string | null) => {
+  // Add support for throttle options set by the API
+  if (throttle && findIndex(['value', throttle], THROTTLE_OPTIONS) < 0) {
+    return [...THROTTLE_OPTIONS, { value: throttle, text: throttle }];
+  }
+
+  return THROTTLE_OPTIONS;
+};
```

Comment:
Awesome! Thanks for fixing this -- our API users will be happy that the UI doesn't reset their custom throttle settings now 😅